scripts/ci/render_bench.py

```diff
@@ -0,0 +1,380 @@
+                        **y_scale,
+                    },
+                }
+            },
+        }
+
+    with open(os.path.join(SCRIPT_PATH, "templates/benchmark.html")) as template_file:
+        html = template_file.read()
+        html = html.replace("%%TITLE%%", title)
+        # double encode to escape the data as a single string
+        html = html.replace('"%%CHARTS%%"', json.dumps(json.dumps(chartjs)))
+    return html
+
+
+class Target(Enum):
+    CRATES = "crates"
+    SIZE = "sizes"
+    ALL = "all"
+
+    def __str__(self) -> str:
+        return self.value
+
+    def includes(self, other: Target) -> bool:
+        return self is Target.ALL or self is other
+
+    def render(self, gcs: storage.Client, end_date: datetime, num_days: int) -> dict[str, str]:
+        commits_by_date = get_commits(end_date, num_days)
+        out: dict[str, str] = {}
+        if self.includes(Target.CRATES):
+            data = get_crates_benchmark_data(gcs, commits_by_date)
+            out[str(Target.CRATES)] = render_html("Rust Crates", data)
+        if self.includes(Target.SIZE):
+            data = get_size_benchmark_data(gcs, commits_by_date)
+            out[str(Target.SIZE)] = render_html("Sizes", data)
+        return out
+
+
+def date_type(v: str) -> datetime:
+    try:
+        return datetime.strptime(v, DATE_FORMAT)
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"Date must be in {DATE_FORMAT} format")
+
+
+def days_type(v: Any) -> int:
+    try:
+        num_days = int(v)
+        if num_days < 1:
+            raise argparse.ArgumentTypeError(f"number of days must be greater than 1, got {v}")
+        return num_days
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"number of days must be a valid integer, got {v}")
+
+
+class Output(Enum):
+    STDOUT = "stdout"
+    GCS = "gcs"
+    FILE = "file"
+
+    def parse(o: str) -> Output:
+        if o == "-":
+            return Output.STDOUT
+        if o.startswith("gs://"):
+            return Output.GCS
+        return Output.FILE
+
+
+@dataclass
+class GcsPath:
+    bucket: str
+    blob: str
+
+
+def parse_gcs_path(path: str) -> GcsPath:
+    if not path.startswith("gs://"):
+        raise ValueError(f"invalid gcs path: {path}")
+    path = path.lstrip("gs://")
+    try:
+        bucket, blob = path.split("/", 1)
+        return GcsPath(bucket, blob.rstrip("/"))
+    except ValueError:
+        raise ValueError(f"invalid gcs path: {path}")
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Render benchmarks from data in GCS",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument("target", type=Target, choices=list(Target), help="Which benchmark to render")
+    parser.add_argument(
+        "--end-date",
+        type=date_type,
+        help=f"The last date to fetch, in {ESCAPED_DATE_FORMAT} format. Default: today ({datetime.today().strftime(DATE_FORMAT)})",
+    )
```

Comment:
If we're just fetching _every_ commit instead of one per day, then the git log could just be `--after=<today - 30 days>`, so it would definitely catch the commit that's currently checked out.